dynawo/sources/Modeler/Common/DYNSubModel.cpp

```diff
@@ -969,6 +963,175 @@ SubModel::addParameterCurve(shared_ptr<curves::Curve>& curve) {
   curve->setNegated(false);
 }
 
+void
+SubModel::printLocalInitParametersValues() const {
+  const boost::unordered_map<std::string, ParameterModeler>& params = getParametersDynamic();
+  std::set<std::string> sortedParams;
+  for (boost::unordered_map<std::string, ParameterModeler>::const_iterator it = params.begin(), itEnd = params.end();
+      it != itEnd; ++it) {
+    sortedParams.insert(it->first);
+  }
+  if (!params.empty()) {
+    Trace::debug(Trace::parameters()) << "------------------------------" << Trace::endline;
+    Trace::debug(Trace::parameters()) << "SubModel " << name()  << " parameters after local initialization"<< Trace::endline;
+    Trace::debug(Trace::parameters()) << "------------------------------" << Trace::endline;
+  }
+
+  for (std::set<std::string>::const_iterator it = sortedParams.begin(), itEnd = sortedParams.end(); it != itEnd; ++it) {
+    const ParameterModeler& parameter = params.find(*it)->second;
+    if (parameter.originSet() && parameter.getOrigin() != LOCAL_INIT)
+      continue;
+    if (!parameter.hasValue()) {
+      continue;
+    }
+    switch (parameter.getValueType()) {
+      case VAR_TYPE_BOOL: {
+        const bool value = parameter.getValue<bool>();
+        Trace::debug(Trace::parameters()) << DYNLog(ParamValueInOrigin, *it, origin2Str(parameter.getOrigin()), value) << Trace::endline;
+        break;
+      }
+      case VAR_TYPE_INT: {
+        const int value = parameter.getValue<int>();
+        Trace::debug(Trace::parameters()) << DYNLog(ParamValueInOrigin, *it, origin2Str(parameter.getOrigin()), value) << Trace::endline;
+        break;
+      }
+      case VAR_TYPE_DOUBLE: {
+        const double& value = parameter.getValue<double>();
+        Trace::debug(Trace::parameters()) << DYNLog(ParamValueInOrigin, *it, origin2Str(parameter.getOrigin()), value) << Trace::endline;
+        break;
+      }
+      case VAR_TYPE_STRING: {
+        const string& value = parameter.getValue<string>();
+        Trace::debug(Trace::parameters()) << DYNLog(ParamValueInOrigin, *it, origin2Str(parameter.getOrigin()), value) << Trace::endline;
+        break;
+      }
+      default:
+      {
+        throw DYNError(Error::MODELER, ParameterNoTypeDetected, *it);
+      }
+    }
+  }
+
+  for (std::set<std::string>::const_iterator it = sortedParams.begin(), itEnd = sortedParams.end(); it != itEnd; ++it) {
+    const ParameterModeler& parameter = params.find(*it)->second;
+    if (!parameter.hasValue()) {
+      if (parameter.isFullyInternal()) {
+        continue;
+      } else {
+        Trace::debug(Trace::parameters()) << DYNLog(ParamNoValueFound, *it) << Trace::endline;
+      }
+    }
+  }
+}
+
+
+void
+SubModel::printParameterValues() const {
+  const boost::unordered_map<std::string, ParameterModeler>& initParams = getParametersInit();
```

Comment:
If it was me I would merge all these parameters in one section `parameters before local initialization` It would make more sense for me, given the obtained list of parameters... What do you think?

Reply:
If you think we loose useful info, don't consider my comment...

Reply:
Do you mean merging init and dynamic parameters values?

Reply:
before init yes, in order to have two sections, one before init, one after, without distinction dynamic/init (reading the log file and given what I am writting in the model, I don't always understand the distinction made here)...

Reply:
The idea is to to display the initial model param (e.g. P/Q0Pu) used to do the local initialization as they are not available in the dynamic model.  We could merge this but there might be duplicated parameters names (similarly to to the generator model)

Reply:
ok fine, we don't want duplicated parameters